dandi/metadata.py

```diff
@@ -71,6 +128,20 @@ def get_metadata(path: Union[str, Path]) -> Optional[dict]:
             lgr.debug("Failed to get metadata for %s: %s", path, exc)
             return None
 
+    # Somewhat less fragile search than previous proposals,
+    # could still be augmented with `_is_nwb` to disambiguate both cases
+    # at the detection level.
+    if _path_in_bids(path):
+        from .bids_validator_xs import validate_bids
+
+        _meta = validate_bids(path)
+        meta = _meta["match_listing"][0]
```

Comment:
note: referencings like `[0]` 
- make it not clear what / why we are getting the first element only
- source of trouble since would crash if list/tuple is empty
- sooner than later we should overhaul that `validate_bids` to provide a result which makes returned value "more obvious"

Reply:
agree, in principle that `[0]` is not nice, but (a) possibly a fix for another PR, (b) it actually makes sense here. A file name could be matched by 2 patterns. That in turn could be an issue with the schema (in which case QA should be done in upstream BIDS, not here), or it could be having `participants.json` *and* `participants.tsv` — not entirely sure whether BIDS agrees that this is broken, and even if it is probably not a reason to invalidate any of the files specifically. Either or both could contain valid metadata.